api/jobs.go

```diff
@@ -1244,6 +1247,10 @@ type JobPlanResponse struct {
 	// Warnings contains any warnings about the given job. These may include
 	// deprecation warnings.
 	Warnings string
+
+	// SchedulerWarnings contains any warnings generated by the scheduler that
+	// are not necessaryly related to any placement or task groups.
+	SchedulerWarnings string
```

Comment:
Let's make this `[]string`. I think a flat string above was a mistake. The names imply multiple warnings, but if it's not a slice we have to figure out a format that works well for the CLI as well as programmatic API users. So if there can be multiple let's just use a slice.